opus/application/test_api/api_test_helper.py

```diff
@@ -166,3 +168,29 @@ def _run_csv_equal_file(self, url, exp_file):
         print('Expected:')
         print(expected)
         self.assertEqual(resp, expected)
+
+    def _run_zip_equal(self, url, expected, response_type='json'):
+        print(url)
+        response = self._get_response(url)
+        self.assertEqual(response.status_code, 200)
+        if response_type == 'json':
+            jdata = json.loads(response.content)
+            file = jdata['filename']
+            filename = file.replace(settings.TAR_FILE_URL_PATH, '')
+            path = settings.TAR_FILE_PATH + filename
+            zip_file = zipfile.ZipFile(path, mode='r')
+        elif response_type == 'binary':
+            binary_stream = BytesIO(response.content)
+            zip_file = zipfile.ZipFile(binary_stream, mode='r')
+        resp = zip_file.namelist()
```

Comment:
If you're going to use a series of "if...elif" then there needs to be something to catch the last clause, like an "else: assert False" or something. Otherwise if you pass in the wrong argument it will simply crash with an undefined variable. Alternatively you can just replace the "elif" with a plain "else".